src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -18,6 +18,12 @@ private class WorkspaceHost : ForegroundThreadAffinitizedObject, IVisualStudioWo
             private readonly VisualStudioWorkspaceImpl _workspace;
             private readonly RemoteHostClient _client;
 
+            /// <summary>
+            /// A <see cref="RemoteHostClient.Connection"/> to use if non-null.  If null, we'll create 
+            /// our own connection to the OOP process.
+            /// </summary>
+            private Connection _preferredConnection;
```

Comment:
:bulb: Using `ReferenceCountedDisposable<Connection>` here would mean the caller wouldn't need to check a `dispose` field and apply conditional logic.